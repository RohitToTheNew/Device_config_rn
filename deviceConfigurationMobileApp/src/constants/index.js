import UUIDMappingMS700 from './uuidMapping-ms700';
import UUIDMappingInfoView from './uuidMapping-infoview';
import DefaultCredentials from './defaultCredentials';
const boudRate = [9600, 19200, 38400, 57600, 115200]

export {DefaultCredentials, UUIDMappingMS700, UUIDMappingInfoView, boudRate};
