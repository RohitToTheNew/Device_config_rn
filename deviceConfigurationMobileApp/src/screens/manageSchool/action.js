const identifire = 'manage-school-log: ';
import { Alert } from 'react-native';
import RNFS from 'react-native-fs';
import Mailer from 'react-native-mail';
import Utils from '../../utils';
import { translate } from '../../translations/translationHelper';

/**
 * function to get saved Schools data from local storage
 */
function getSavedData() {
  const result = RNFS.readFile(RNFS.DocumentDirectoryPath + '/test.txt', 'utf8');
  return result;
}

/**
 * function for formatting the rooms data to desired CSV format
 * @param {object} parms object containing the data of rooms
 * @returns formatted CSV data
 */
function getDataInDesiredFormate(parms) {
  let CSVData = [];
  parms.forEach(room => {
    for (const key in room) {
      if (key === 'devices') {
        room[key].forEach(roomData => {
          let obj = {};
          obj.driverName = roomData.deviceType;
          obj.deviceName = roomData.deviceName;
          obj.ip = roomData.ipAddress || '';
          obj.mac = roomData.macAddress;
          obj.roomName = room['title'];
          CSVData.push(obj);
        });
      }
    }
  });
  return CSVData;
}

/**
 * function to save schools list to local storage
 * @param {object} schools object containing schools list
 * @returns 
 */
const writeDataTolocalStorage = async schools => {
  const data = JSON.stringify(schools);
  const path = RNFS.DocumentDirectoryPath + '/test.txt';
  try {
    await RNFS.writeFile(path, data, 'utf8');
    return `FILE WRITTEN!`;
  } catch (err) {
    Utils.Log(Utils.logType.error, 'error in writeDataTolocalStorage', err)
    return `Error while writing`;
  }
};

/**
 * function to write schools data to local storage
 * @param {object} schools object containing data of schools
 * @param {string} fileName name of file containing data of schools
 * @returns error(if any) during wiriting to local storage OR success message if writing is successful
 */

const writeCSVDataTolocalStorage = async (schools, fileName) => {
  const resultCSV = convertToCSV(schools);
  const pathToWrite = `${RNFS.DocumentDirectoryPath}/${fileName}.csv`;
  try {
    await RNFS.writeFile(pathToWrite, resultCSV, 'utf8');
    return `FILE WRITTEN!`;
  } catch (err) {
    Utils.showToast(translate('errorWhileWriting.'));
    return `Error while writing`;
  }
};
/**
 * function to attach generated CSV in email and open email client
 * @param {string} fileName name of file containing data of schools
 */
function sendCSVAsMail(schoolName, fileName) {
  try {
    Mailer.mail(
      {
        subject: `${schoolName} Devices`,
        isHTML: true,
        attachments: [
          {
            type: 'csv',
            path: `${RNFS.DocumentDirectoryPath}/${fileName}.csv`, // The absolute path of the file from which to read data.
          },
        ],
      },

      (error, event) => {
        Alert.alert(
          error,
          event,
          [
            {
              text: 'Ok',
              onPress: () => { },
            },
            {
              text: 'Cancel',
              onPress: () => { },
            },
          ],
          { cancelable: true },
        );
      },
    );
  } catch (error) {
    Utils.Log(Utils.logType.error, 'error in sendCSVAsMail', error);
  }
}

/**
 * function to handle the 'Export CSV' action for school
 * @param {object} parms object containig data of school
 */
async function handleExportCSV(parms) {
  try {
    const schoolName = parms?.schoolTitle;

    let month = new Date().getMonth() + 1;
    let date = new Date().getDate();
    const year = new Date().getFullYear();

    if (date < 10) date = '0' + date;
    if (month < 10) month = '0' + month;

    const hour = new Date().getHours();
    const minute = new Date().getMinutes();
    const second = new Date().getSeconds();

    const currentDate = `${month}-${date}-${year}`;
    const currentTime = `${hour}H${minute}M${second}S`;

    const fileName = `${schoolName}_Devices_${currentDate}_${currentTime}`
      .toString()
      .trim();
    const CSVData = getDataInDesiredFormate(parms?.rooms);
    await writeCSVDataTolocalStorage(CSVData, fileName);
    sendCSVAsMail(schoolName, fileName);
  } catch (error) {
    Utils.Log(Utils.logType.error, 'handleExportCSV', error);
  }
}

/**
 * function to convert the object data to desired CSV format
 * @param {object} arr array of rooms details
 * @returns
 */
function convertToCSV(arr) {
  const array = [Object.keys(arr[0])].concat(arr);
  return array
    .map(it => {
      return Object.values(it).toString();
    })
    .join('\n');
}

/**
 * function to show alert when delete button is pressed
 * @param {string} schoolTitle title of school
 * @param {string} schoolId school id
 * @param {function} setSchoolTitle function to set school title
 * @param {function} setSchoolId function to set school id
 * @param {function} setPopupToDeleteDevice function to set popup to show
 */
function deleteSchoolAlert(
  schoolTitle,
  schoolId,
  setSchoolTitle,
  setSchoolId,
  setPopupToDeleteDevice,
) {
  setSchoolTitle(schoolTitle);
  setSchoolId(schoolId);
  setPopupToDeleteDevice(true);
}

/**
 * function to delete school
 * @param {string} schoolID school id
 * @param {object} savedData object containing saved schools data
 * @param {function} setSavedData function to set saved schools data after change
 * @param {*} setPopupToDeleteDevice function to set popup to hide
 */
async function deleteSchool(
  schoolID,
  savedData,
  setSavedData,
  setPopupToDeleteDevice,
) {
  try {
    const filteredData = savedData.filter(school => school?.id !== schoolID);
    await writeDataTolocalStorage(filteredData);
    setSavedData(filteredData);
    setPopupToDeleteDevice(false);
  } catch (error) {
    Utils.Log(Utils.logType.error, 'deleteSchool', error);
  }
}
export {
  getSavedData,
  getDataInDesiredFormate,
  writeCSVDataTolocalStorage,
  sendCSVAsMail,
  handleExportCSV,
  deleteSchoolAlert,
  deleteSchool,
  writeDataTolocalStorage,
};
