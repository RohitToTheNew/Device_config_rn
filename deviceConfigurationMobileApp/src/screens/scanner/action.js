const identifire = 'scanner-action log: ';
import { check, request, RESULTS, PERMISSIONS } from 'react-native-permissions';
import { RNCamera } from 'react-native-camera';
import utils from '../../utils';
import { translate } from '../../translations/translationHelper';
import RNFS from 'react-native-fs';

/**
 * On click Enter details manually and redirect user to Add device screen
 * @param {object} navigation it represents navigation object
 * @returns object
 */
function actionHandleManualDetails(navigation) {
  return () => {
    navigation.navigate('addDevice');
  };
}

/**
 * function to read local file data from storage
 * @returns file details read from storage
 */
const readFileFromLocalStorage = async () => {
  let result = RNFS.readFile(RNFS.DocumentDirectoryPath + '/test.txt', 'utf8');
  return result;
};

/**
 * function to check if the device details encoded into the QR code is having valid keys
 * @param {object} obj object containing the scanned data from QR code
 * @returns boolean indicating weather details are valid or not
 */
const isValidObject = (obj) =>
  obj?.serialNumber && obj?.serialNumber !== '' &&
  obj?.macAddress && obj?.macAddress !== '' &&
  obj?.deviceType && obj?.deviceType !== '';

/**
 * function to check if the device details encoded into the QR code is having valid keys
 * @param {object} obj object containing the scanned data from QR code
 * @returns boolean indicating weather details are valid or not
 */
const isValidDetails = (obj) =>
  obj?.serial && obj?.serial !== '' &&
  obj?.mac && obj?.mac !== '' &&
  obj?.device && obj?.device !== '';

/**
 * On successful scan
 * @param {object} navigation it represents navigation object
 * @param {function} setInvalidCode used to set invalid code boolean value
 * @param {object} e contains scan result
 */
function actionOnSuccess(navigation, setInvalidCode, e) {
  const typeToCheck = utils.isIOS ? 'org.iso.QRCode' : "QR_CODE"
  if (e.type === typeToCheck) {
    try {
      const scannedString = JSON.parse(e?.data);
      console.log('scanned string', scannedString)
      if (isValidObject(scannedString) || isValidDetails(scannedString)) {
        navigation.replace('addDevice', {
          mac: scannedString?.macAddress || scannedString?.mac,
          serialNumber: scannedString?.serialNumber || scannedString?.serial,
          deviceType: scannedString?.deviceType || scannedString?.device,
        });
      } else {
        setInvalidCode(true);
      }
    } catch (e) {
      setInvalidCode(true);
    }
  } else {
    setInvalidCode(true);
  }
}

/**
 * Click on close icon
 * @param {object} navigation it represents navigation object
 * @returns object
 */
function actionHandleClose(navigation) {
  return () => {
    navigation.goBack();
  };
}

/**
 * Click on flash icon
 * @param {boolean} isFlashOn used to identify if flash is on/off
 * @param {function} setFlash used to set flash on/off value
 * @param {function} setIsFlashOn used to set flash on value
 * @returns object
 */
function actionHandleFlash(isFlashOn, setFlash, setIsFlashOn) {
  return () => {
    if (isFlashOn) {
      setFlash(RNCamera.Constants.FlashMode.off);
      setIsFlashOn(false);
    } else {
      setFlash(RNCamera.Constants.FlashMode.torch);
      setIsFlashOn(true);
    }
  };
}

/**
 * To ask camera permission
 * @returns camera permission
 */
async function actionUseEffect() {

  try {
    const result = await check(PERMISSIONS.IOS.CAMERA);
    if (result !== RESULTS.GRANTED) {
      const askCameraPermission = await request(PERMISSIONS.IOS.CAMERA);
      return askCameraPermission;
    } else {
      utils.Log(utils.logType.log, identifire, 'Camera permisson granted');
    }
  } catch (Ex) {
    utils.showToast(translate('error'), translate('permissionsException'));
  }
}
export {
  actionHandleManualDetails,
  actionOnSuccess,
  actionHandleClose,
  actionUseEffect,
  actionHandleFlash,
};
