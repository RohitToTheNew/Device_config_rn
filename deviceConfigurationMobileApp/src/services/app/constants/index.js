export const UPDATE_APP_MODAL_FIELDS = 'UPDATE_APP_MODAL_FIELDS';
