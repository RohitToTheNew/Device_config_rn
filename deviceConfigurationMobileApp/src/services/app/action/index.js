import BleManager from '../../../config/bleManagerInstance';
import { UUIDMappingMS700 } from '../../../constants';
import Utils from '../../../utils';
import {UPDATE_APP_MODAL_FIELDS} from '../constants';
import {sentryErrorHandler} from '../../../utils/errorHandler';
import { translate } from '../../../translations/translationHelper';

export const updateAppModalFields = (key, value) => dispatch => {
  dispatch({
    type: UPDATE_APP_MODAL_FIELDS,
    payload: {[key]: value},
  });
};

/**
 * function that reads POR override value from BLE
 */
export const readPoeOverride = () => {
  return async (dispatch, getState) => {
    const {connectedDevice} = getState().authDevices;
    try {
      const poeValue = await BleManager.readCharacteristicForDevice(
        connectedDevice.id,
        UUIDMappingMS700.rootServiceUDID,
        UUIDMappingMS700.poeOverrride,
      );
      if (Utils.decodeBase64(poeValue.value) == '\u0000\u0000\u0000' || Utils.decodeBase64(poeValue.value) == "" ) {
        dispatch(updateAppModalFields('poeOverride', false)) 
      }else{
        const valueObtained =  Utils.decodeBase64(poeValue.value) == 'true' ? true : false;
        dispatch(updateAppModalFields('poeOverride', valueObtained)) 
      }
    } catch (error) {
      Utils.Log(Utils.logType.error, 'readPoeOverride', error);
      Utils.showToast(translate('unableToReadValuse'));
    }
  };
};

/**
 * redux action that saves POE override settings onto MS700 Device
 * @param {function} callback callback function that gets executed after settings are saved
 */
export const togglePoeOverride = callback => {
  return async (dispatch, getState) => {
    const {connectedDevice} = getState().authDevices;
    const {poeOverride} = getState().app;
    try {
      const valueToUpdate = poeOverride ? "false" : "true"
      await BleManager.writeCharacteristicWithResponseForDevice(
        connectedDevice.id,
        UUIDMappingMS700.rootServiceUDID,
        UUIDMappingMS700.poeOverrride,
        Utils.encodeBase64(valueToUpdate),
      );
      await BleManager.writeCharacteristicWithResponseForDevice(
        connectedDevice.id,
        UUIDMappingMS700.rootServiceUDID,
        UUIDMappingMS700.applyChange,
        Utils.encodeBase64('')
      );
      Utils.showToast('',poeOverride ?'PoE Override disabled' :'PoE Override enabled','success')
      dispatch(updateAppModalFields('poeOverride', !poeOverride))
      callback();
    } catch (error) {
      sentryErrorHandler(`togglePoeOverride eror ${JSON.stringify(error)}`)
      Utils.showToast('Error', 'Not able to change the poe settings, try again.');
    }
  };
};