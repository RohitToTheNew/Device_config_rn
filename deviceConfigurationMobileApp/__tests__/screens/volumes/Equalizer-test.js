import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import {createTestStore} from '../../utilityFunction';
import Equalizer from '../../../src/screens/volumes/equalizer';
import {store} from '../../../src/store/configureStore';
import {render, screen} from '@testing-library/react-native';
import {updateAuthDevices} from '../../../src/services/authDevices/action';
import {
  changeSpeakerImpedence,
  geteqSettingsValues,
  getNoiseSuppressionSettings,
  getSpeakerImpedence,
  toggleBypassEQSetting,
  toggleNoiseSuppressionSetting,
  updateNoiseSuppressionThreshold,
  updateVolumeSettingsFields,
  writeeqSettingsValues,
} from '../../../src/services/volumes/action';
import BleManager from '../../../src/config/bleManagerInstance';
import {UUIDMappingMS700} from '../../../src/constants';
import Toast from 'react-native-toast-message';
import base64 from 'react-native-base64';

describe('on Equalizer screen mounting', () => {
  let tree;
  beforeEach(() => {
    tree = renderer.create(<Equalizer />).toJSON();
  });

  it('renders Equalizer screen correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});

describe('Equalizer settings', () => {
  let component, myStore;
  beforeEach(() => {
    myStore = createTestStore();
    component = (
      <Provider store={myStore}>
        <Equalizer />
      </Provider>
    );
    render(component);
  });
  it('should render equalizer settings list', () => {
    expect(screen.getByTestId('equalizerVolumeSettingsList')).toBeTruthy();
  });
});

describe('geteqSettingsValues function', () => {
  let component, readSpy;
  beforeEach(() => {
    component = (
      <Provider store={store}>
        <Equalizer />
      </Provider>
    );
    render(component);
  });
  beforeAll(async () => {
    store.dispatch(
      updateAuthDevices('connectedDevice', {id: 1, localName: 'Brx-emulator'}),
    );
    readSpy = jest.spyOn(BleManager, 'readCharacteristicForDevice');
  });
  it('should fetch equalizer settings from BLE', async () => {
    await store.dispatch(geteqSettingsValues());
    expect(readSpy).toBeCalledWith(
      1,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.eqBand1,
    );
  });
});

describe('writeeqSettingsValues function with error in writing values on BLE', () => {
  let component, rowData, toastSpy;
  beforeEach(() => {
    component = (
      <Provider store={store}>
        <Equalizer />
      </Provider>
    );
    render(component);
  });
  beforeAll(async () => {
    store.dispatch(updateAuthDevices('connectedDevice', {}));
    rowData = {
      index: 0,
      item: {id: 1, charactersticId: UUIDMappingMS700.eqBand1},
    };
    jest
      .spyOn(BleManager, 'writeCharacteristicWithResponseForDevice')
      .mockImplementation(() => {
        throw new Error('cannot write values because BLE is shutdown.');
      });
    toastSpy = jest.spyOn(Toast, 'show');
  });
  it('should get error from BLE when writing equalizer settings on BLE', async () => {
    await store.dispatch(writeeqSettingsValues(rowData, [30]));
    expect(toastSpy).toBeCalled();
  });
});

describe('writeeqSettingsValues function', () => {
  let component, rowData, writeSpy;
  beforeEach(() => {
    component = (
      <Provider store={store}>
        <Equalizer />
      </Provider>
    );
    render(component);
  });
  beforeAll(async () => {
    store.dispatch(
      updateAuthDevices('connectedDevice', {id: 1, localName: 'Brx-emulator'}),
    );
    rowData = {
      index: 0,
      item: {id: 1, charactersticId: UUIDMappingMS700.eqBand1},
    };
    writeSpy = jest.spyOn(
      BleManager,
      'writeCharacteristicWithResponseForDevice',
    );
    jest.spyOn(base64, 'encode').mockImplementation(() => {
      return 30;
    });
  });
  it('should update equalizer settings values on the BLE', async () => {
    await store.dispatch(writeeqSettingsValues(rowData, [30]));
    expect(writeSpy).toBeCalledWith(
      1,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.eqBand1,
      30,
    );
  });
});

describe('getNoiseSuppressionSettings function', () => {
  let component, readSpy;
  beforeEach(() => {
    component = (
      <Provider store={store}>
        <Equalizer />
      </Provider>
    );
    render(component);
  });
  beforeAll(async () => {
    store.dispatch(
      updateAuthDevices('connectedDevice', {id: 1, localName: 'Brx-emulator'}),
    );
    readSpy = jest.spyOn(BleManager, 'readCharacteristicForDevice');
  });
  it('should fetch noise suppression settings from BLE', async () => {
    await store.dispatch(getNoiseSuppressionSettings());
    expect(readSpy).toBeCalledWith(
      1,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.noiseSuppression,
    );
    expect(readSpy).toBeCalledWith(
      1,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.noiseSuppressionThreshold,
    );
  });
});

describe('getSpeakerImpedence function', () => {
  let component, readSpy;
  beforeEach(() => {
    component = (
      <Provider store={store}>
        <Equalizer />
      </Provider>
    );
    render(component);
  });
  beforeAll(async () => {
    store.dispatch(
      updateAuthDevices('connectedDevice', {id: 1, localName: 'Brx-emulator'}),
    );
    readSpy = jest.spyOn(BleManager, 'readCharacteristicForDevice');
  });
  it('should fetch speaker impedence settings from BLE', async () => {
    await store.dispatch(getSpeakerImpedence());
    expect(readSpy).toBeCalledWith(
      1,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.speakerImpedence,
    );
  });
});

describe('updateNoiseSuppressionThreshold function', () => {
  let component, writeSpy;
  beforeEach(() => {
    component = (
      <Provider store={store}>
        <Equalizer />
      </Provider>
    );
    render(component);
  });
  beforeAll(async () => {
    store.dispatch(
      updateAuthDevices('connectedDevice', {id: 1, localName: 'Brx-emulator'}),
    );
    writeSpy = jest.spyOn(
      BleManager,
      'writeCharacteristicWithResponseForDevice',
    );
    jest.spyOn(base64, 'encode').mockImplementation(() => {
      return 30;
    });
  });
  it('should update noise suppression threshold setting value on the BLE', async () => {
    await store.dispatch(updateNoiseSuppressionThreshold([30]));
    expect(writeSpy).toBeCalledWith(
      1,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.noiseSuppressionThreshold,
      30,
    );
  });
});

describe('toggleNoiseSuppressionSetting function', () => {
  let component, callback, writeSpy;
  beforeEach(() => {
    component = (
      <Provider store={store}>
        <Equalizer />
      </Provider>
    );
    render(component);
  });
  beforeAll(async () => {
    store.dispatch(
      updateAuthDevices('connectedDevice', {id: 1, localName: 'Brx-emulator'}),
    );
    callback = jest.fn();
    writeSpy = jest.spyOn(
      BleManager,
      'writeCharacteristicWithResponseForDevice',
    );
    jest.spyOn(base64, 'encode').mockImplementation(() => {
      return 'false';
    });
  });
  it('should toggle noise suppression enabled setting value on the BLE', async () => {
    await store.dispatch(toggleNoiseSuppressionSetting(callback));
    expect(writeSpy).toBeCalledWith(
      1,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.noiseSuppressionThreshold,
      'false',
    );
  });
});

describe('changeSpeakerImpedence function', () => {
  let component, writeSpy;
  beforeEach(() => {
    component = (
      <Provider store={store}>
        <Equalizer />
      </Provider>
    );
    render(component);
  });
  beforeAll(async () => {
    store.dispatch(
      updateAuthDevices('connectedDevice', {id: 1, localName: 'Brx-emulator'}),
    );
    writeSpy = jest.spyOn(
      BleManager,
      'writeCharacteristicWithResponseForDevice',
    );
    jest.spyOn(base64, 'encode').mockImplementation(() => {
      return '8';
    });
  });
  it('should update speaker impedence setting value on the BLE', async () => {
    await store.dispatch(changeSpeakerImpedence('8'));
    expect(writeSpy).toBeCalledWith(
      1,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.speakerImpedence,
      '8',
    );
    });

  describe('toggleBypassEQSetting function', () => {
    let component, writeSpy;
    beforeEach(() => {
      component = (
        <Provider store={store}>
          <Equalizer />
        </Provider>
      );
      render(component);
    });
    beforeAll(async () => {
      store.dispatch(
        updateAuthDevices('connectedDevice', {
          id: 1,
          localName: 'Brx-emulator',
        }),
      );
      writeSpy = jest.spyOn(
        BleManager,
        'writeCharacteristicWithResponseForDevice',
      );
    });
    it('should toggle bypass EQ Setting on the BLE, when byPass is enabled', async () => {
      const callbackFunction = jest.fn();
      jest.spyOn(base64, 'encode').mockImplementation(() => {
        return true;
      });
      await store.dispatch(toggleBypassEQSetting(callbackFunction));
      expect(writeSpy).toBeCalledWith(
        1,
        UUIDMappingMS700.rootServiceUDID,
        UUIDMappingMS700.byPassEQ,
        true,
      );
      expect(writeSpy).toBeCalledWith(
        1,
        UUIDMappingMS700.rootServiceUDID,
        UUIDMappingMS700.applyChange,
        true,
      );
    });
    it('should toggle bypass EQ Setting on the BLE, resets the EQ band values to 0, when byPass is disabled', async () => {
      store.dispatch(updateVolumeSettingsFields('bypassEQ', true));
      const callbackFunction = jest.fn();
      jest
        .spyOn(base64, 'encode')
        .mockImplementationOnce(() => {
          return '0';
        })
        .mockImplementationOnce(() => {
          return '0';
        })
        .mockImplementationOnce(() => {
          return '0';
        })
        .mockImplementationOnce(() => {
          return '0';
        })
        .mockImplementationOnce(() => {
          return '0';
        })
        .mockImplementationOnce(() => {
          return true;
        });
      await store.dispatch(toggleBypassEQSetting(callbackFunction));
      expect(writeSpy).toBeCalledWith(
        1,
        UUIDMappingMS700.rootServiceUDID,
        UUIDMappingMS700.eqBand1,
        '0',
      );
      expect(writeSpy).toBeCalledWith(
        1,
        UUIDMappingMS700.rootServiceUDID,
        UUIDMappingMS700.eqBand2,
        '0',
      );
      expect(writeSpy).toBeCalledWith(
        1,
        UUIDMappingMS700.rootServiceUDID,
        UUIDMappingMS700.eqBand3,
        '0',
      );
      expect(writeSpy).toBeCalledWith(
        1,
        UUIDMappingMS700.rootServiceUDID,
        UUIDMappingMS700.eqBand4,
        '0',
      );
      expect(writeSpy).toBeCalledWith(
        1,
        UUIDMappingMS700.rootServiceUDID,
        UUIDMappingMS700.eqBand5,
        '0',
      );
      expect(writeSpy).toBeCalledWith(
        1,
        UUIDMappingMS700.rootServiceUDID,
        UUIDMappingMS700.byPassEQ,
        true,
      );
      expect(writeSpy).toBeCalledWith(
        1,
        UUIDMappingMS700.rootServiceUDID,
        UUIDMappingMS700.applyChange,
        true,
      );
    });
  });
});
