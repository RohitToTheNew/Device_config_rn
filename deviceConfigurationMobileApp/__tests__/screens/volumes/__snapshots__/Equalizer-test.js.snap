// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`on Equalizer screen mounting renders Equalizer screen correctly 1`] = `null`;
